### ISSUE TITLE
Subquery in Where Clause Not Parsed

### ISSUE THREAD

**Issue by reata** [reata/sqllineage#204]
```sql
SELECT col1
FROM tab1
WHERE col1 IN (SELECT max(col1) FROM tab2);
```

Source table is expected to include tab1 and tab2 while it only says tab1 in v1.3.2

This is regression since v1.3.0 when we introduced subquery parsing logic. Used to work correctly in v1.2.4

**Comment by TrololoLi** [comment]
Hi! Sorry for jumping in, but I am not sure whether it's a bug or a feature =)  If I see tab2 as a source for col1 in this query I would be confused.  We don't bring any new data from tab2, we only filter on it...

**Comment by reata** [comment]
> Hi! Sorry for jumping in, but I am not sure whether it's a bug or a feature =) If I see tab2 as a source for col1 in this query I would be confused. We don't bring any new data from tab2, we only filter on it...

I might not have made it clear. Let me rephrase: suppose we insert the result from this query to tab3. Then at table level, the source table of tab3 will be tab1 and tab2. But at column level, the source column of tab3.col1 will just be tab1.col1 because as you said, tab2 is used for filtering purpose.

Some times, sqllineage is not used for data engineering purposes. System administrator may just need to get the tables accessed/read in the query, and decide whether or not authorize the query based on their ACL. I want to get this use case right.
